src/data/set/opposite.lean

```diff
@@ -0,0 +1,59 @@
+/-
+Copyright (c) 2022 Markus Himmel. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Markus Himmel
+-/
+import data.opposite
+import data.set.basic
+
+/-!
+# The opposite of a set
+
+The opposite of a set `s` is simply the set obtained by taking the opposite of each member of `s`.
+-/
+
+variables {α : Type*}
+
+open opposite
+
+namespace set
+
+/-- The opposite of a set `s` is the set obtained by taking the opposite of each member of `s`. -/
+protected def op (s : set α) : set αᵒᵖ :=
+op '' s
+
+/-- The unop of a set `s` is the set obtained by taking the unop of each member of `s`. -/
+protected def unop (s : set αᵒᵖ) : set α :=
+unop '' s
```

Comment:
Using the preimage has much nicer properties.
```suggestion
/-- The opposite of a set `s` is the set obtained by taking the opposite of each member of `s`. -/
protected def op (s : set α) : set αᵒᵖ := unop ⁻¹' s

/-- The unop of a set `s` is the set obtained by taking the unop of each member of `s`. -/
protected def unop (s : set αᵒᵖ) : set α := op ⁻¹' s
```
For example, you can then replace all the `by tidy` with `rfl`

Reply:
You should also bundle them as equivalences.
```suggestion
/-- The opposite of a set `s` is the set obtained by taking the opposite of each member of `s`. -/
protected def op (s : set α) : set α ≃ set αᵒᵖ :=
⟨set.preimage unop, set.preimage op, sorry, sorry⟩

/-- The unop of a set `s` is the set obtained by taking the unop of each member of `s`. -/
protected def unop : set αᵒᵖ ≃ set α :=
⟨set.preimage op, set.preimage unop, sorry, sorry⟩
```

Reply:
I used the preimage, but bundling it in the way you suggested partially breaks dot notation, so I decided to state the eqivalence separately.